client/my-sites/email/email-management/home/email-plan-warnings.tsx

```diff
@@ -21,17 +28,45 @@ type EmailPlanWarningsProps = {
 
 const EmailPlanWarnings = ( { domain, emailAccount }: EmailPlanWarningsProps ) => {
 	const translate = useTranslate();
-	const selectedSiteSlug = useSelector( getSelectedSiteSlug );
+	const selectedSite = useSelector( getSelectedSite );
+	const selectedSiteSlug = selectedSite?.slug ?? '';
 
 	const warning = emailAccount?.warnings?.[ 0 ];
 
+	const cannotAddEmailWarningReason = getCurrentUserCannotAddEmailReason( domain );
+	const cannotAddEmailWarningMessage = cannotAddEmailWarningReason?.message ?? '';
+	const cannotAddEmailWarningCode = cannotAddEmailWarningReason?.code ?? null;
+
+	const WarningMessageForCode = useMemo( () => {
+		return ( props: { code: string | number | null } ) => {
+			switch ( props.code ) {
+				case EMAIL_WARNING_CODE_OTHER_USER_OWNS_EMAIL:
+					return (
+						<EmailNonOwnerMessage
+							domainName={ domain.name }
+							selectedSite={ selectedSite }
+							source="email-management"
+						/>
+					);
+				case EMAIL_WARNING_CODE_OTHER_USER_OWNS_DOMAIN_SUBSCRIPTION:
+					return (
+						<EmailNonDomainOwnerMessage
+							domain={ domain }
+							source={ 'email-management' }
+							usePromoCard={ false }
+							selectedSite={ selectedSite }
+						/>
+					);
+				default:
+					return null;
```

Comment:
`cannotAddEmailWarningMessage` could contain information about other semi-esoteric states of the domain. I'm wondering if we shouldn't render that message in the default case here.